Stack/Opc.Ua.Core/Stack/Tcp/TcpServerChannel.cs

```diff
@@ -682,7 +682,14 @@ private bool ProcessOpenSecureChannelRequest(uint messageType, ArraySegment<byte
                 State = TcpChannelState.Open;
 
                 // send the response.
-                SendOpenSecureChannelResponse(requestId, CurrentToken, request);
```

Comment:
the fix is good, simpler were to get ahold of the response token..
var responseToken = token
...
SendOpenSecureChannelResponse(requestId, responseToken, request);